src/main/thrift/parquet.thrift

```diff
@@ -779,8 +812,12 @@ struct ColumnMetaData {
    * whether we can decode those pages. **/
   2: required list<Encoding> encodings
 
-  /** Path in schema **/
-  3: required list<string> path_in_schema
+  /** Path in schema 
+    *  Example of deprecated a field for PAR3
```

Comment:
Obviously I'm a bit biased, but I find the approach of defining separate structs (`FileMetadataV3`, etc.) much cleaner than tediously documenting what is required in V1 and forbidden in V3, and vice-versa.

Reply:
I think there are trade-offs I replied on the mailing list with more details (hopefully we can centralize discussion there).